### PR TITLE
Add separate job for api version upgrade specs

### PR DESCRIPTION
**Pull request by shysank**

### CODE REVIEW

**Comment by shysank** [comment]
assign @CecileRobertMichon

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 1]
```diff
@@ -538,6 +538,47 @@ presubmits:
...
+              value: "API Version Upgrade"
+          # we need privileged mode in order to do docker in docker
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 7300m
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
+      testgrid-tab-name: capz-pr-e2e-apiversion-upgrade
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:581]:
```suggestion
      testgrid-tab-name: capz-pr-e2e-apiversion-upgrade-main
```

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 2]
```diff
@@ -538,6 +538,47 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: capz-pr-e2e-upgrade-1-21-1-22
+  - name: pull-cluster-api-provider-azure-e2e-apiversion-upgrade-v1beta1
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:541]: is this running both alpha3 -> beta1 and alpha4 -> beta1 ?

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:541]:
```suggestion
  - name: pull-cluster-api-provider-azure-apiversion-upgrade
```

Comment (shysank) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:541]: yes

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 3]
```diff
@@ -538,6 +538,47 @@ presubmits:
...
+              value: "API Version Upgrade"
+          # we need privileged mode in order to do docker in docker
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 7300m
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
+      testgrid-tab-name: capz-pr-e2e-apiversion-upgrade-main
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:581]:
```suggestion
      testgrid-tab-name: capz-pr-apiversion-upgrade-main
```

it's not actually running e2e tests on second thought

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 4]
```diff
@@ -538,6 +538,47 @@ presubmits:
...
+            - name: GINKGO_FOCUS
+              value: "API Version Upgrade"
+          # we need privileged mode in order to do docker in docker
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 7300m
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:580]: can we add a description that explains what this does?

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**